app/public/components/filter/filteringPanel.js

```diff
@@ -0,0 +1,138 @@
+
+        const leftFilterType = document.getElementById('left-filter-placeholder').innerHTML;
+        const rightFilterType = document.getElementById('right-filter-placeholder').innerHTML;
+
+        const leftFilterInput = document.getElementById(leftInputId).value;
+        const rightFilterInput = document.getElementById(rightInputId).value;
+
+        if (rightFilterInput) {
+            model.filtering.addFilter(fieldNameValue, rightFilterInput, rightFilterType);
+        }
+
+        if (leftFilterInput) {
+            model.filtering.addFilter(fieldNameValue, leftFilterInput, leftFilterType);
+        }
+
+        clearUserInput();
+    };
+
+    const fieldSelect = h('select.select', {
+        id: fieldSelectId,
+        name: 'showOptions' }, [
+        h('option', {
+            value: '',
+            selected: true,
+            disabled: true,
+            hidden: true,
+        }, 'Field'),
+        fields.map((field) => h('option', { value: field.name }, field.fieldName)),
+    ], h('.close-10-primary'));
+
+    const typeSelect = h('select.select', {
+        id: typeSelectId,
+        name: 'showOptions',
+        onchange: () => onFilteringTypeChange(),
+    }, [
+        h('option', { value: '', selected: true, disabled: true, hidden: true }, 'Filtering type'),
+        h('option', { value: aggregatedFiltersTypes }, aggregatedFiltersTypes),
+        h('option', { value: filterTypes.between }, filterTypes.between),
+    ], h('.close-10-primary'));
+
+    const leftInput = h('.text-field',
+        h('input.form-control.relative', {
+            style: 'width:120px',
+            type: 'text',
+            value: document.getElementById(typeSelectId)?.options[Selection.selectedIndex]?.value,
+            disabled: false,
+            id: leftInputId,
+            required: true,
+        }),
+        h('span.placeholder', { id: 'left-filter-placeholder' }, 'match/from'));
+
+    const rightInput = h('.text-field',
+        h('input.form-control.relative', {
+            style: 'width:120px',
+            type: 'text',
+            value: '',
+            disabled: false,
+            id: rightInputId,
+            required: true,
+        }),
+        h('span.placeholder', { id: 'right-filter-placeholder' }, 'to/exclude'));
+
+    return h('.font-size-small', [
+        h('.flex-wrap.justify-between.items-center',
+            h('.flex-wrap.justify-between.items-center',
+                h('h5', 'Filter data'),
+                h('button.btn.btn-secondary', {
+                    onclick: () => {},
+                }, 'Use defined filters'))),
+        h('.flex-wrap.justify-between.items-center',
+            h('.flex-wrap.justify-between.items-center',
+                fieldSelect,
+                typeSelect,
+                leftInput,
+                rightInput,
+
+                h('button.btn.btn-primary', {
+                    onclick: () => onFilterSubmit(),
+                }, 'Filter'),
```

Comment:
anti-pattern: abstraction levels combination

Reply:
Moved to separate variables.